app/helpers/concerns/puppetdb_foreman/hosts_helper_extensions.rb

```diff
@@ -0,0 +1,24 @@
+module PuppetdbForeman
+  module HostsHelperExtensions
+    extend ActiveSupport::Concern
+
+    module Overrides
+      def show_appropriate_host_buttons(host)
+        buttons = super
+
+        buttons << link_to_if_authorized(
+          _('Puppet Classes'),
+          hash_for_puppetdb_foreman_node_path(:id => host.name),
+          :title => _('Browse host puppet classes'),
+          :class => 'btn btn-default'
+        )
```

Comment:
Can we add a guard for this to only display the button if the host has puppet enabled? Maybe something like `if host.puppet_proxy_id?`. It's not perfect, but it saves a query to PuppetDB to check if the host is present there.

Reply:
The guard you suggested works perfectly :)